lib/src/connector/connector.cc

```diff
@@ -722,6 +722,10 @@ void Connector::stopMonitorTaskAndWait() {
     } else {
         LOG_WARNING("The Monitoring Thread is not joinable");
     }
+
+    if (monitor_exception_) {
+        std::rethrow_exception(monitor_exception_);
```

Comment:
This could be a bad idea, as it can be called during the destructor for `Connector`.